.github/general_coding_instructions.md

```diff
@@ -0,0 +1,798 @@
+# Azure Messaging Block Storage - General Coding Guidelines
+
+## Overview {#overview}
+This document establishes coding standards and conventions for the Azure Messaging Block Storage (EBS) project. These guidelines ensure consistency, maintainability, and alignment with Azure C library patterns used throughout the codebase.
+
+## Function Naming Conventions {#function-naming}
+
+### API Function Naming
+- Use **lowercase with underscores** (snake_case) for all function names
+- **NEVER use camelCase, PascalCase, or any mixed-case naming**
+- Follow the pattern: `<module_prefix>_<action>[_<qualifier>]`
+- Examples:
+  - `bsdl_address_list_create()`
+  - `block_storage_append_async()`
+  - `msi_token_cache_get_token()`
+  - `operation_limiter_acquire()`
+
+### Internal Function Naming
+- **Internal functions must be declared `static`** and not exposed through public headers
+- **Prefix internal functions with the module name and `internal`**:
+  - Pattern: `<module_prefix>_internal_<action>[_<qualifier>]`
+  - Examples:
+    - `static int bsdl_internal_validate_parameters(...)`
+    - `static void bs_internal_cleanup_resources(...)`
+    - `static RESULT_TYPE sf_internal_process_callback(...)`
+- This clearly distinguishes internal implementation details from public API
+
+### Module Prefixes
+- Use consistent module prefixes that identify the component:
+  - `bsdl_*` - Block Storage Data Layer functions
+  - `bs_*` - Block Storage functions
+  - `sf_*` - Service Fabric related functions
+  - `zrpc_*` - ZRPC framework functions
```

Comment:
This section is a guideline, but I'm not sure if it's a MUST. ZRPC has been moving away from this,.  Geo follows it maybe 80%.

Reply:
I would leave it in for the moment. I'm trying to get this in c build tools so that we see how MrBot behaves now in various repos. I am hesitant so start tweaking too much, so I will "won't fix" this one if you don't feel horrible strong about it.